os-autoinst-obs-auto-submit

```diff
@@ -16,7 +16,24 @@ XMLSTARLET=$(command -v xmlstarlet || true)
 [[ -n $XMLSTARLET ]] || (echo "Need xmlstarlet" && exit 1)
 
 # shellcheck source=/dev/null
-. "$(dirname "$0")"/_common
+. "$(dirname "${BASH_SOURCE[0]}")"/_common
+
+check_or_update_submit_target_extra_version() {
+    # extract :Update
+    update_prj="${submit_target_extra##*:}"
+    # extract openSUSE:Backports:SLE-15-SPx
+    current_prj="${submit_target_extra%:*}"
+    # extact the actual version to compare
+    current_prj="${current_prj%:*}"
+    # awk iterates through all the fields in the match to avoid fixed position
+    latest_submit_target_extra_version=$(osc dists | grep "$current_prj" | awk -v prj="$current_prj" '{for (i=1; i<=NF; i++) if ($i ~ "^" prj ":") print $i}' | sort --version-sort | tail -n 1)
+    if [[ -z $latest_submit_target_extra_version ]]; then
+        echo "Project not found." && return 1
+    fi
+    echo "$latest_submit_target_extra_version:$update_prj"
+}
+
+submit_target_extra=check_or_update_submit_target_extra_version
```

Comment:
In line 10 a default is set allowing callers to set a custom value but here you overwrite it again. That's not intuitive. I think you should define a project root as default like `submit_target_extra_root=${submit_target_extra_root:-"openSUSE:Backports:"}"`, use that and then only as default set `submit_target_extra` to the output of check_or_update_submit_target_extra_version

Reply:
Please read my suggestion again. You now still overwrite any potentially user specific config settings

Reply:
I would say that it updates the user input to actually use the latest version. it can be either ways. you can check again

Reply:
It does not make sense to make submit_target_extra a user specifiable variable but then overwrite it

Reply:
I agree user-specified variables shouldn't be re-used. Even if it works it's confusing to follow.

How about `submit_target_extra_latest_version`

Reply:
are you talking about `submit_target_extra` or `submit_target_extra_project`?

Reply:
Well, this was about `submit_target_extra=check_or_update_submit_target_extra_version`.

I guess now that you are renaming the variable and only using it internally that no longer applies.


Please update the commit/PR description to clarify, though, since that still talks about the old name 🤔 Something like **Pull in latest version of projects with submit_target_extra_project** if that is what it does?

Reply:
I guess you still didn't read or understand my suggestion from https://github.com/os-autoinst/scripts/pull/390#discussion_r2073407164

Please do

```bash
submit_target_extra="${submit_target_extra:-check_or_update_submit_target_extra_version}"
```